cogs/odds_on_cog.py

```diff
@@ -0,0 +1,110 @@
+#!/usr/bin/env python
+# -*- coding: utf-8 -*-
+"""
+An example cog to show how things should be done.
+
+Also provides a simple base for starting a new cog.
+"""
+# In this case, discord import is not needed, in some cases it may be.
+import asyncio
+
+import discord
+from discord.ext import commands
+import utils as ut
+
+
+class OddsOnCog(commands.Cog):
+    """Create a class that extends Cog to make our functionality in."""
+
+    def __init__(self, bot):
+        """Save our bot argument that is passed in to the class."""
+        self.bot = bot
+
+    @commands.command(
+        name="oddson",
+        help="Let's you play Odds On against other members")
+    @commands.has_role("Member")
+    async def odds_on(self, ctx, role, *, wager: str):
+        """
+        Challenge a user to odds on
+
+        Challenge a specific user to a wager,
+        that user then specifies the odds and
+        both users then pick a number between 1
+        and the chosen upper limit.
+        """
+        target_user_id = await ut.find_id(role)
+        if target_user_id == ctx.author.id:
+            raise commands.errors.UserInputError()
+        target_user = self.bot.get_user(target_user_id)
+        if target_user is None:
+            raise commands.errors.UserInputError(message="I couldn't find the user you tagged.")
+
+        confirm_message = await ctx.send(f"<@{target_user_id}>, <@{ctx.author.id}> has challenged you in odds on.\n"
+                                         f"Reply to this message with a number (e.g. 7) to set the odds or ignore it.")
+
+        def check(check_message):
+            try:
+                return int(check_message.content) > 1 and check_message.author == target_user
+            except ValueError:
+                return False
+
+        try:
+            message = await self.bot.wait_for('message', timeout=30.0, check=check)
+        except asyncio.TimeoutError:
+            await confirm_message.delete()
+            await ctx.send(f"<@{ctx.author.id}> The person you challenged did not respond in time.")
```

Comment:
Might be better to `return` at the end of the `except` clause to avoid the extra level of indentation caused by `else`